### PR TITLE
wp-parsley: Some loader improvements & maybe hide plugin UI

### PR DESCRIPTION
**Pull request by jblz**
## Description

We want to enable the plugin for some sites but not show the UI (settings menu / page, admin header warning, widget & controls). This change unhooks certain actions and filters when the plugin is loaded by this wrapper file _*unless*_ it was enabled via the `wpvip_parsely_load_mu` filter.

For future extensibility sake, we also check the return value of a new filter `wpvip_parsely_hide_ui_for_mu` which can be independently overridden.

A bit of refactoring is also included:
* namespace the wrapper file to `Automattic\VIP\WP_Parsely_Integration`
* simplify the function name since it's namespaced
* pull the list of `SUPPORTED_VERSIONS` out to a constant
* attempt to load versions in top-down order (the first element is the default, so no need to maintain the `WPVIP_PARSELY_DEFAULT_VERSION` constant.
* ensure the entry file is readable prior to `require`ing the path
* rename the `wpvip_parsely_major_version` filter to `wpvip_parsely_version` -- it's not used anywhere and the new format is more compact and correct

## Changelog Description

Plugin Updated: VIP Parse.ly Integration -- Hide plugin UI in some cases

We made some behind-the-scenes changes that will help customers try Parse.ly on their sites more easily.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1. Check out PR.
1. Ensure a proper value for the `apikey` in the `parsely` blog option (you can use the settings page in the plugin if it's active, otherwise, you can do something like `wp option patch insert parsely apikey example.com`)
1. Clear the activation option: `wp option delete _wpvip_parsely_mu`
1. Ensure you're not adding a filter to `wpvip_parsely_load_mu` in any plugin
  - The plugin should not be loaded at all -- no tracker, no meta, no admin UI
1. Set the activation option: `wp option set _wpvip_parsely_mu 1`
  - The plugin should be loaded (tracker & meta in the src), but no UI should be visible in WP-Admin > Settings > Parsely
1. Add the following to a sourced plugin file: `add_filter( 'wpvip_parsely_load_mu', '__return_true' );`
  - The plugin should be loaded & UI should be visible & fully functional in WP-Admin > Settings > Parsely
1. Clear the activation option: `wp option delete _wpvip_parsely_mu`
  - The plugin should be loaded & UI should be visible & fully functional in WP-Admin > Settings > Parsely
1. Add the following to a sourced plugin file: `add_filter( 'wpvip_parsely_hide_ui_for_mu', '__return_true' );`
  - The plugin should be loaded (tracker & meta in the src), but no UI should be visible in WP-Admin > Settings > Parsely

### CODE REVIEW

File: wp-parsely.php [hunk 1]
```diff
@@ -34,20 +39,41 @@ function wpvip_load_wp_parsely_plugin() {
...
-	] ) ) {
+	$specified_version = apply_filters( 'wpvip_parsely_version', false );
+
+	if ( in_array( $specified_version, SUPPORTED_VERSIONS ) ) {
+		array_unshift( $versions_to_try, $specified_version );
+		$versions_to_try = array_unique( $versions_to_try );
+	} else {
 		trigger_error(
-			sprintf( 'Invalid value configured via wpvip_parsely_major_version filter: %s', $major_version ),
+			sprintf( 'Invalid value configured via wpvip_parsely_version filter: %s', $version ),
```
Comment (jblz) [inline wp-parsely.php:55]: Not sure why this is hitting `WordPress.Security.EscapeOutput.OutputNotEscaped` in phpcs -- I'm not outputting anything 🤔 

https://app.circleci.com/pipelines/github/Automattic/vip-go-mu-plugins/6713/workflows/3faa430e-6757-45d4-ab8d-6d847fc0d4a6/jobs/46847

Comment (GaryJones) [reply wp-parsely.php:55]: You are as part of the `trigger_error()`.

See https://github.com/WordPress/WordPress-Coding-Standards/blob/dc2f21771cb2b5336a7e6bb6616abcdfa691d7de/WordPress/Sniffs/Security/EscapeOutputSniff.php#L198

**Comment by jblz** [comment]
Looks like the nightly tests are broken in other recent PRs, so seems unrelated to this change.

**Review by pauarge — APPROVED** [review]
LGTM

@jblz We're looking at why tests are failing on nightly, as you point out, it has nothing to do with your changes.

**Pull request merged**